en/ros/ros2_offboard_control.md

```diff
@@ -1,172 +1,162 @@
-     receive: true
-   ...
-   - msg:     vehicle_local_position_setpoint
-     receive: true
-   - msg:     trajectory_setpoint # multi-topic / alias of vehicle_local_position_setpoint
-     base:    vehicle_local_position_setpoint
-     receive: true
+## Trying it out
+
+Follow the instructions in [ROS 2 User Guide](..ros/ros2_comm.md) to install PX and run the simulator, install ROS2, and start the XRCE-DDS Agent.
+After that we can follow a similar set of steps to those in [ROS 2 User Guide > Build ROS2 Workspace](..ros/ros2_comm.md#build-ros-2-workspace) to run the example.
+
+To build and run the example:
+
+1. Open a new terminal.
+1. Create and navigate into a new colcon workspace directory using:
+
+   ```sh
+   mkdir -p ~/ws_offboard_control/src/
+   cd ~/ws_offboard_control/src/
    ```
 
-   In *path_to_colcon_ws/src/px4_ros_com/templates/urtps_bridge_topics.yaml*:
-   ```yaml
-   - msg:     OffboardControlMode
-     receive: true
-   ...
-   - msg:     VehicleCommmand
-     receive: true
-   ...
-   - msg:     VehicleLocalPositionSetpoint
-     receive: true
-   - msg:     TrajectorySetpoint
-     base:    VehicleLocalPositionSetpoint
-     receive: true
+1. Clone the [px4_msgs](https://github.com/PX4/px4_msgs) repo to the `/src` directory (this repo is needed in every ROS2 PX4 workspace!):
+
+   ```sh
+   git clone https://github.com/PX4/px4_msgs.git
+   # checkout the matching release branch if not using PX4 main.
    ```
- 1. `vehicle_command` message is configured in the `urtps_bridge_topics.yaml` file both in the PX4-Autopilot and
- *px4_ros_com* package to *send* to the Autopilot.
 
-    In *PX4-Autopilot/msg/tools/urtps_bridge_topics.yaml*:
-    ```yaml
-    - msg:     vehicle_command
-      receive: true
-    ```
+1. Clone the example repository [px4_ros_com](https://github.com/PX4/px4_ros_com) to the `/src` directory:
 
-    In *path_to_colcon_ws/src/px4_ros_com/templates/uorb_rtps_message_ids.yaml*:
-    ```yaml
-    - msg:     VehicleCommmand
-      receive: true
-    ```
+   ```sh
+   git clone https://github.com/PX4/px4_ros_com.git
+   ```
 
-   :::note
-   At time of writing, the above topic is already configured to be sent.
-   :::
+1. Source the ROS2 development environment ("foxy") into the current terminal and compile the workspace using `colcon`:
 
-## Implementation
+   ```sh
+   cd ..
+   source /opt/ros/foxy/setup.bash
+   colcon build
+   ```
 
-The source code of the offboard control example can be found in [offboard_control.cpp](https://github.com/PX4/px4_ros_com/blob/main/src/examples/offboard/offboard_control.cpp).
+1. Source the `local_setup.bash`:
 
-Here are some details about the implementation:
+   ```sh
+   source install/local_setup.bash
+   ```
+1. Launch the example.
 
-```cpp
-timesync_sub_ = this->create_subscription<px4_msgs::msg::Timesync>("/fmu/timesync/out",
-    10,
-    [this](const px4_msgs::msg::Timesync::UniquePtr msg) {
-        timestamp_.store(msg->timestamp);
-    });
-```
```

Comment:
This wasn't in the source code, which just seems to get the time to send from the local clock. I believe ROS2 does everything it needs to sync "under the hood"?